esda/local_join_count.py

```diff
@@ -0,0 +1,162 @@
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator
+from libpysal import weights
+from esda.crand import (
+    crand as _crand_plus,
+    njit as _njit,
+    _prepare_univariate
+)
+
+
+PERMUTATIONS = 999
+
+
+class Local_Join_Count(BaseEstimator):
+
+    """Univariate Local Join Count Statistic"""
+
+    def __init__(self, connectivity=None, permutations=PERMUTATIONS, n_jobs=1, 
+                 keep_simulations=True, seed=None):
+        """
+        Initialize a Local_Join_Count estimator
+        Arguments
+        ---------
+        connectivity     : scipy.sparse matrix object
+                           the connectivity structure describing
+                           the relationships between observed units.
+                           Need not be row-standardized.
+        permutations     : int
+                           number of random permutations for calculation of pseudo
+                           p_values
+        n_jobs           : int
+                           Number of cores to be used in the conditional randomisation. If -1,
+                           all available cores are used.    
+        keep_simulations : Boolean
+                           (default=True)
+                           If True, the entire matrix of replications under the null 
+                           is stored in memory and accessible; otherwise, replications 
+                           are not saved
+        seed             : None/int
+                           Seed to ensure reproducibility of conditional randomizations. 
+                           Must be set here, and not outside of the function, since numba 
+                           does not correctly interpret external seeds 
+                           nor numpy.random.RandomState instances.              
+                           
+        Attributes
+        ----------
+        LJC             : numpy array
+                          array containing the univariate
+                          Local Join Count (LJC).
+        p_sim           : numpy array
+                          array containing the simulated
+                          p-values for each unit.
+
+        """
+
+        self.connectivity = connectivity
+        self.permutations = permutations
+        self.n_jobs = n_jobs
+        self.keep_simulations = keep_simulations
+        self.seed = seed
+
+    def fit(self, y, n_jobs=1, permutations=999):
```

Comment:
If you define permutations above, then you should probably not set the default here again! 

Also `sklearn` estimators generally put these kinds of options in the init, not fit. And, all `sklearn` estimators, even the unsupervised ones, [take both an X and a y](https://scikit-learn.org/stable/developers/develop.html#fitting)... we should follow those conventions, I think.

Reply:
Fixed permutations such that it is part of `init(...)` and removed constant from top of file. Applied to all LJC functions. See https://github.com/pysal/esda/pull/139/commits/2eb0d37d059af4596a2ffd0e103820a7ae0dcd74.